packages/react-ui/src/widgets/comparative/ComparativeFormulaWidgetUI.js

```diff
@@ -58,93 +56,93 @@ function ComparativeFormulaWidgetUI({
   const theme = useTheme();
   const classes = useStyles();
 
-  function getColor(index) {
-    return colors[index] || {};
-  }
-  function getLabel(index) {
-    return labels[index] || {};
-  }
+  const processedData = useMemo(
+    () =>
+      data
+        .map((d, i) => ({
+          ...d,
+          color: getColor(colors, i)
+        }))
+        .filter((d) => d.value !== undefined),
```

Comment:
You're doing here first .map and then .filter, vs previous .filter > .map. Is it intended @aaranadev?

Reply:
Yes, we need to remove the undefined values, but we need to assign the correct color index before the element is removed it.

Example:

If the second value is undefined, we need the value after the first to have the third color and not the second.